Classes/Manager/Submanagers/OCTSubmanagerAvatars.m

```diff
@@ -18,19 +22,85 @@ @implementation OCTSubmanagerAvatars
 
 - (void)setAvatar:(UIImage *)avatar
 {
-    // TODO
+    id <OCTFileStorageProtocol> storage = [self.dataSource managerGetFileStorage];
+    NSString *path = [storage.pathForAvatarsDirectory stringByAppendingPathComponent:userAvatarFileName];
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+
+    if ([fileManager fileExistsAtPath:path])
+        [fileManager removeItemAtPath:path error:nil];
+
+    OCTTox *tox = [self.dataSource managerGetTox];
+    NSData *data = nil;
+
+    if (avatar) {
+        data = [self pngDataFromImage:avatar];
+        [fileManager createDirectoryAtPath:[path stringByDeletingLastPathComponent]
+               withIntermediateDirectories:YES
+                                attributes:nil
+                                     error:nil];
+        [data writeToFile:path atomically:NO];
+    }
+    
+    [tox setAvatar:data];
 }
 
 - (UIImage *)avatar
 {
-    // TODO
+    id <OCTFileStorageProtocol> storage = [self.dataSource managerGetFileStorage];
+    NSString *path = [storage.pathForAvatarsDirectory stringByAppendingPathComponent:userAvatarFileName];
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+
+    if ([fileManager fileExistsAtPath:path]) {
+        NSData *data = [NSData dataWithContentsOfFile:path];
+        UIImage *avatar = [UIImage imageWithData:data];
+        return avatar;
+    }
     return nil;
 }
 
 - (BOOL)hasAvatar
 {
-    // TODO
-    return NO;
+    id <OCTFileStorageProtocol> storage = [self.dataSource managerGetFileStorage];
+    NSString *path = [storage.pathForAvatarsDirectory stringByAppendingPathComponent:userAvatarFileName];
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+
+    return ([fileManager fileExistsAtPath:path]);
+}
+
+#pragma mark - Private Method
+
+- (NSData *)pngDataFromImage:(UIImage *)image
```

Comment:
Also it would be nice to test this method. Maybe you can [create fake image](https://stackoverflow.com/a/993159) with huge size and then check if the returned NSData was properly resized (if it is smaller than maximum data length).